sale_order_revision/model/sale_order.py

```diff
@@ -64,6 +69,17 @@ def copy_quotation(self):
         msg = _('New revision created: %s') % self.name
         self.message_post(body=msg)
         old_revision.message_post(body=msg)
+
+        # set stored procurement group id on old order
+        old_revision.write({'procurement_group_id': procurement_group_id})
+
+        # swap order lines of old and new order
+        so_line = self.env['sale.order.line']
+        old_lines = so_line.browse(old_revision.order_line.ids)
```

Comment:
You don't need to rebrowse, just assign the recordset: `old_lines = old_revision.order_line`